### PR TITLE
Add networksecurityperimeter and managedidentitywithdelegation common types 

### PR DESCRIPTION
**Pull request by AlitzelMendez**
issue: https://github.com/Azure/typespec-azure/issues/1160

### CODE REVIEW

File: packages/samples/common-types/openapi/v5/networksecurityperimeter.json
Comment (timotheeguerin) [inline packages/samples/common-types/openapi/v5/networksecurityperimeter.json:67]: This one I think used to be inline
image

maybe better if for now we can keep things structure(not introduce new definitions) as they were unless its not possible

File: packages/samples/common-types/openapi/v5/networksecurityperimeter.json
Comment (timotheeguerin) [inline packages/samples/common-types/openapi/v5/networksecurityperimeter.json:190]: There is a diff here 
image
good to investigate which one is correct

File: packages/samples/common-types/openapi/v5/networksecurityperimeter.json
Comment (timotheeguerin) [inline packages/samples/common-types/openapi/v5/networksecurityperimeter.json:234]: This wasn't there before
image

File: packages/samples/common-types/openapi/v5/networksecurityperimeter.json
Comment (AlitzelMendez) [reply packages/samples/common-types/openapi/v5/networksecurityperimeter.json:186]: type is the correct one ([AutoRest Extensions for OpenAPI 2.0 | autorest (azure.github.io)](https://azure.github.io/autorest/extensions/#x-ms-arm-id-details)) thanks for your help!!!

File: packages/samples/common-types/openapi/v5/networksecurityperimeter.json
Comment (AlitzelMendez) [reply packages/samples/common-types/openapi/v5/networksecurityperimeter.json:230]: I tried to address this one but I am unsure if it is a problem on my definition or when the generation of the common types is happening.

For what I saw, "x-ms-identifiers" is included when there is not an id for the array, in this case there is an id coming from resource: https://github.com/Azure/typespec-azure/blob/80f55767e0ff73699de4cba05b754529f8a5fc14/packages/typespec-azure-resource-manager/lib/common-types/network-security-perimeter.tsp#L170 (is not directly in there but it is pulled) even if I tried to add an id, I am not able to do so because it is already inherited from the other model, but when the common-type is getting generated it includes the x-ms-identifiers like it didn't have the id, am I missing something?

Comment (markcowl) [reply packages/samples/common-types/openapi/v5/networksecurityperimeter.json:230]: I think this is related to this bug: https://github.com/Azure/typespec-azure/issues/1358  as the same logic is likely to be used

If so, we probably want to fix this before we merge this PR

File: packages/typespec-azure-resource-manager/lib/common-types/network-security-perimeter.tsp
Comment (markcowl) [reply packages/typespec-azure-resource-manager/lib/common-types/network-security-perimeter.tsp:11]: Do we need to add this to other common-types definitions for correct swagger geneeration?

File: packages/typespec-azure-resource-manager/lib/common-types/network-security-perimeter.tsp
Comment (markcowl) [inline packages/typespec-azure-resource-manager/lib/common-types/network-security-perimeter.tsp:84]: I don't see this in the common-types in the spec repo

File: packages/typespec-azure-resource-manager/lib/common-types/network-security-perimeter.tsp
Comment (markcowl) [inline packages/typespec-azure-resource-manager/lib/common-types/network-security-perimeter.tsp:176]:
```suggestion
  @OpenAPI.extension("x-ms-identifiers", ["name"])
```

File: packages/typespec-azure-resource-manager/lib/common-types/network-security-perimeter.tsp
Comment (markcowl) [inline packages/typespec-azure-resource-manager/lib/common-types/network-security-perimeter.tsp:263]: We need to decide how to handle the segment name in this parameter model.  We might wanto to do this: https://github.com/Azure/typespec-azure/issues/1358

Or there may be a static value that is always be used.  This depends on usage inside specs

File: packages/typespec-azure-resource-manager/lib/common-types/network-security-perimeter.tsp
Comment (markcowl) [inline packages/typespec-azure-resource-manager/lib/common-types/network-security-perimeter.tsp:265]: Are the common-types decoration (-ref.tsp) files being added later?

**Review by markcowl — CHANGES_REQUESTED** [review]

File: packages/typespec-azure-resource-manager/lib/common-types/network-security-perimeter.tsp
Comment (AlitzelMendez) [reply packages/typespec-azure-resource-manager/lib/common-types/network-security-perimeter.tsp:176]: I think this one is empty: https://github.com/Azure/azure-rest-api-specs/blob/34f5146bc945549d087d38a8a593c0a5f475ad7f/specification/common-types/resource-management/v5/networksecurityperimeter.json#L107

File: packages/typespec-azure-resource-manager/lib/common-types/network-security-perimeter.tsp
Comment (AlitzelMendez) [reply packages/typespec-azure-resource-manager/lib/common-types/network-security-perimeter.tsp:11]: I don't think we need it, I added it to have the more similar generation to this: https://github.com/Azure/typespec-azure/blob/c469c8b695f2d5c412c2b9998ba5a14badac9e51/packages/samples/common-types/openapi/v5/networksecurityperimeter.json

File: packages/typespec-azure-resource-manager/lib/common-types/network-security-perimeter.tsp
Comment (AlitzelMendez) [reply packages/typespec-azure-resource-manager/lib/common-types/network-security-perimeter.tsp:263]: Mark, I think that is the link for the bug that you mentioned above, could you please share the link for the suggestion?

File: packages/typespec-azure-resource-manager/lib/common-types/network-security-perimeter.tsp
Comment (markcowl) [reply packages/typespec-azure-resource-manager/lib/common-types/network-security-perimeter.tsp:166]: Yeah, it probably should not be, but we should match

File: packages/typespec-azure-resource-manager/lib/common-types/network-security-perimeter.tsp
Comment (markcowl) [reply packages/typespec-azure-resource-manager/lib/common-types/network-security-perimeter.tsp:253]: Yes, I meant a template, like this: https://github.com/Azure/typespec-azure/blob/95cc6b95355aac89280f39bc4b5b9ae94f6ae16d/packages/typespec-azure-resource-manager/lib/common-types/private-links.tsp#L117-L127

**Review by markcowl — CHANGES_REQUESTED** [review]

**Review by markcowl — APPROVED** [review]
Approved pending resolution of the two remaining issues

File: packages/typespec-azure-resource-manager/lib/common-types/network-security-perimeter.tsp
Comment (timotheeguerin) [inline packages/typespec-azure-resource-manager/lib/common-types/network-security-perimeter.tsp:6]: should also remove info here, and augment it in the samples file instead

**Pull request merged**
